### PR TITLE
azure: parse ProvisionGuestProxyAgent as bool

### PR DESCRIPTION
**Pull request by KsenijaS**
Parse ProvisionGuestProxyAgent  property as bool instead of string.

### CODE REVIEW

File: tests/unittests/sources/test_azure.py [hunk 1]
```diff
@@ -2835,6 +2835,14 @@ def test_read_azure_ovf_with_proxy_guest_agent(self):
         cfg = ret[2]
         self.assertTrue(cfg["ProvisionGuestProxyAgent"])
```
Comment (cjp256) [inline tests/unittests/sources/test_azure.py:2836]:
```suggestion
        self.assertTrue(cfg["ProvisionGuestProxyAgent"] is True)
```

File: tests/unittests/sources/test_azure.py [hunk 2]
```diff
@@ -2835,6 +2835,14 @@ def test_read_azure_ovf_with_proxy_guest_agent(self):
...
 
+    def test_read_azure_ovf_with_proxy_guest_agent(self):
+        """The read_azure_ovf method should set ProvisionGuestProxyAgent
+        cfg flag to True."""
+        content = construct_ovf_env(provision_guest_proxy_agent=False)
+        ret = dsaz.read_azure_ovf(content)
+        cfg = ret[2]
+        self.assertTrue(cfg["ProvisionGuestProxyAgent"] == False)
```
Comment (cjp256) [inline tests/unittests/sources/test_azure.py:2844]:
```suggestion
        assert cfg["ProvisionGuestProxyAgent"] is False
```

File: tests/unittests/sources/test_azure.py [hunk 3]
```diff
@@ -2835,6 +2835,14 @@ def test_read_azure_ovf_with_proxy_guest_agent(self):
         cfg = ret[2]
         self.assertTrue(cfg["ProvisionGuestProxyAgent"])
 
+    def test_read_azure_ovf_with_proxy_guest_agent(self):
```
Comment (cjp256) [inline tests/unittests/sources/test_azure.py:2838]: test name duplicates previous test, please consolidate with parametrize

**Review by cjp256 — APPROVED** [review]
LGTM, thanks!

**Review by holmanb — APPROVED** [review]

**Pull request merged**